lib/path.dart

```diff
@@ -120,7 +120,7 @@ Uri? _currentUriBase;
 ///
 /// This is cached because [current] is called frequently but rarely actually
 /// changes.
-String? _current;
+late final String _current;
```

Comment:
Bad nullable assumption leaks in all over! – From here!

Reply:
This is a little suspicious - does this really never get overwritten?  The "rarely changed" suggests otherwise.

Reply:
Yup – had to go back to `String?`

Reply:
Does late (but not final) work?  It seems like it should, unless there's a dynamic assignment to this.

Reply:
Nope. Need the `null` check above.